analyzer/tools/build-logger/tests/unit/test_paths.py

```diff
@@ -0,0 +1,357 @@
+                [binary], env=empty_env, outs="--VARIABLE--"
+            )
+            self.assert_json(
+                command=f"{cc} {file} -Werror -o {binary}",
+                file=file,
+            )
+
+    def test_compiler_path2(self):
+        if not AVAILABLE_GNU_COMPILERS:
+            self.skipTest(
+                f"No compiler matches the 'g++-*' pattern in your PATH"
+            )
+
+        for cc in AVAILABLE_GNU_COMPILERS:
+            self.tearDown()  # Cleanup the previous iteration.
+            self.setUp()
+            logger_env = self.get_envvars()
+            logger_env["CC_LOGGER_GCC_LIKE"] = "/g++-"
+            file = self.source_file
+            binary = self.binary_file
+            self.assume_successful_command(
+                [cc, file, "-Werror", "-o", binary], logger_env
+            )
+            self.assume_successful_command(
+                [binary], env=empty_env, outs="--VARIABLE--"
+            )
+            actual_json = self.read_actual_json()
+            self.assertEqual(actual_json, "[\n]")
+
+    def test_simple(self):
+        logger_env = self.get_envvars()
+        file = self.source_file
+        binary = self.binary_file
+        self.assume_successful_command(
+            ["g++", file, "-Werror", "-o", binary], logger_env
+        )
+        self.assume_successful_command(
+            [binary], env=empty_env, outs="--VARIABLE--"
+        )
+        self.assert_json(
+            command=f"/usr/bin/g++ {file} -Werror -o {binary}",
+            file=file,
+        )
+
+    def test_cpath(self):
+        logger_env = self.get_envvars()
+        logger_env["CPATH"] = "path1"
+        file = self.source_file
+        binary = self.binary_file
+        self.assume_successful_command(
+            ["g++", file, "-Werror", "-o", binary], logger_env
+        )
+        self.assume_successful_command(
+            [binary], env=empty_env, outs="--VARIABLE--"
+        )
+        self.assert_json(
+            command=f"/usr/bin/g++ -I path1 {file} -Werror -o {binary}",
+            file=file,
+        )
+
+    def test_cpath_after_last_I(self):
+        logger_env = self.get_envvars()
+        logger_env["CPATH"] = ":path1:path2:"
+        file = self.source_file
+        binary = self.binary_file
+        self.assume_successful_command(
+            [
+                "g++",
+                "-I",
+                "p0",
+                file,
+                "-Werror",
+                "-o",
+                binary,
+                "-I",
+                "p1",
+                "-I",
+                "p2",
+            ],
+            logger_env,
+        )
+        self.assume_successful_command(
+            [binary], env=empty_env, outs="--VARIABLE--"
+        )
+        self.assert_json(
+            command=f"/usr/bin/g++ -I p0 {file} -Werror -o {binary} -I p1 "
+            f"-I p2 -I . -I path1 -I path2 -I .",
+            file=file,
+        )
+
+    def test_cplus(self):
```

Comment:
I think the names of the tests should be more descriptive. E.g. what is the difference between `test_cplus` and `test_cpp`? Or could you please add a descriptive comment for each test case?

Reply:
Yes, but I'm not intending to change their names. This way it's easier to map the old versions to the new versions.
One could do the renaming in a follow-up commit but I don't think it would be that useful anyway.